### ISSUE TITLE
[BottomNavigation] accessibilityValue is not being updated when badgeValue is updated

### ISSUE THREAD

**Issue by romoore** [material-components/material-components-ios#4733]
The BottomNavigation ItemView's accessibilityValue is expected to default to the badgeValue. When set custom, the custom value should be used instead.  However, setting the badge value twice results in the accessibilityValue failing to update.

## Reproduction steps

```objectivec
- (void)testDefaultAccessibilityValueUpdatesWhenBadgeValueChanges {
  // Given
  MDCBottomNavigationItemView *itemView = [[MDCBottomNavigationItemView alloc] init];

  // When
  itemView.badgeValue = @"777 apples";
  itemView.badgeValue = @"65 blueberries";

  // Then
  XCTAssertEqualObjects(itemView.accessibilityValue, @"65 blueberries");
}
```
